packages/core/src/elements-context.ts

```diff
@@ -107,3 +177,61 @@ function mergeElementsForContext(
   }
   return result;
 }
+
+function removeHiddenFields(
+  elements: Record<string, InternalSchemaElement>,
+  accessPolicyResource: AccessPolicyResource | undefined,
+  keyPrefix?: string
+): Record<string, InternalSchemaElement> {
+  if (!accessPolicyResource?.hiddenFields?.length) {
+    return elements;
+  }
+
+  const prefix = keyPrefix ? keyPrefix + '.' : '';
+  return Object.fromEntries(
+    Object.entries(elements).filter(([key]) => !matchesKeyPrefixes(prefix + key, accessPolicyResource.hiddenFields))
+  );
+}
+
+function markReadonlyFields(
```

Comment:
Just tagging this as a pretty important detail of the implementation.  The in-memory schema is now mutable, and a function of the user's access policy.

I'm trying to think through the implications of that...  For example, in a multi profile setting, if I change my profile, and therefore change my access policy, does that invalidate the cache of structure definitions and `ExtendedInternalSchemaElement` values?

Reply:
I considered that and should have added a comment/test about it since it is quite important, but the in-memory schema isn't being mutated since the element object is spread when setting `readonly` and the `result` object is a newly create object as well. So unless I'm missing something, this should be fine? I'm still going to add a test asserting as much to make sure that doesn't drift over time.

```js
    if (isReadonly) {
      result[key] = { ...element, readonly: true };
    } else {
      result[key] = element;
    }
```